pastas/timeseries.py

```diff
@@ -740,26 +523,192 @@ def to_dict(self, series: Optional[bool] = True) -> dict:
+    2. Make sure the index is a DatetimeIndex
+    3. Make sure the indices are datetime64
+    4. Make sure the index is monotonically increasing
+    5. Make sure there are no duplicate indices
+    6. Make sure the time series has no nan-values
+
+    If any of these checks are not passed the method will throw an error that needs
+    to be fixed by the user.
+
+    Examples
+    --------
+
+    >>> ps.validate_oseries(series)
+
+    """
+    _validate_series(series, equidistant=False)
+
+
+def _validate_series(series: Series, equidistant: bool = True):
+    """Internal method to validate user-provided input time series.
+
+    Parameters
+    ----------
+    series: pandas.Series
+        Pandas.Series object containing the series time series.
+    equidistant: bool, optional
+        Whether the time series should have equidistant time step or not.
+
+    Notes
+    -----
+    If any of these checks are not passed the method will throw an error that needs
+    to be fixed by the user.
+
+    """
+    # Because we are friendly and allow 1D DataFrames
+    if isinstance(series, pd.DataFrame):
+        if len(series.columns) == 1:
+            series = series.iloc[:, 0]
+
+    # 0. Make sure it is a Series and not something else (e.g., DataFrame)
+    if not isinstance(series, pd.Series):
+        msg = f"Expected a Pandas Series, got {type(series)}"
+        logger.error(msg)
+        raise ValueError(msg)
+
+    name = series.name  # Only Series have a name, DateFrame do not
+
+    # 1. Make sure the values are floats
+    if not pd.api.types.is_float_dtype(series):
+        msg = f"Values of time series {name} are not dtype=float."
+        logger.error(msg)
+        raise ValueError(msg)
+
+    # 2. Make sure the index is a DatetimeIndex
+    if not isinstance(series.index, pd.DatetimeIndex):
+        msg = f"Index os series {name} is not a pandas.DatetimeIndex."
+        logger.error(msg)
+        raise ValueError(msg)
+
+    # 3. Make sure the indices are datetime64
+    if not pd.api.types.is_datetime64_dtype(series.index):
+        msg = f"Indices os series {name} are not datetime64."
+        logger.error(msg)
+        raise ValueError(msg)
+
+    # 4. Make sure the index is monotonically increasing
+    if not series.index.is_monotonic_increasing:
+        msg = (
+            f"The time-indices of series {name} are not monotonically increasing. Try "
+            f"to use `series.sort_index()` to fix it."
+        )
+        logger.error(msg)
+        raise ValueError(msg)
+
+    # 5. Make sure there are no duplicate indices
+    if not series.index.is_unique:
+        msg = (
+            f"duplicate time-indexes were found in the time series {name}. Make sure "
+            f"there are no duplicate indices. For example by "
+            f"`grouped = series.groupby(level=0); series = grouped.mean()`"
+        )
+        logger.error(msg)
+        raise ValueError(msg)
+
+    # 6. Make sure the time series has no nan-values
```

Comment:
For oseries NaNs are dropped automatically, do we add a flag checking for NaNs in oseries? Or do we prefer people dropping nans from the oseries prior to passing it to Model?

Reply:
Different philosophies exist here I think. I am in favour of requiring dropping NaNs by users, but it an easy operation like 1D DataFrames so I included it :)